### PR TITLE
replace all occurrences of @tagless annotation with the manual impl

### LINKED ISSUE #292: Migrate @tagless annotations
We've a couple of `@tagless` annotations that we'll need to migrate for #290.

### PR DESCRIPTION
**Pull request by pepegar**
~Currently, most of the implementaitions only needed the `Handler[F]`
type in the companion.~ Not even that :)

FIXES #292

### CODE REVIEW

File: modules/examples/todolist/client/src/main/scala/clients/TodoListClient.scala [hunk 1]
```diff
@@ -36,4 +33,8 @@ trait TodoListClient[F[_]] {
 
   def remove(id: Int): F[Int]
 
-}
\ No newline at end of file
+}
+
+object TodoListClient {
+  trait Handler[G[_]] extends TodoListClient[G]
```
Comment (diesalbla) [inline modules/examples/todolist/client/src/main/scala/clients/TodoListClient.scala]: Same as in [my other comment](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/examples/todolist/client/src/main/scala/clients/TodoItemClient.scala [hunk 2]
```diff
@@ -35,4 +33,8 @@ trait TodoItemClient[F[_]] {
 
   def remove(id: Int): F[Int]
 
-}
\ No newline at end of file
+}
+
+object TodoItemClient {
+  trait Handler[G[_]] extends TodoItemClient[G]
```
Comment (diesalbla) [inline modules/examples/todolist/client/src/main/scala/clients/TodoItemClient.scala]: Same as in [my other comment](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/examples/todolist/client/src/main/scala/clients/TagClient.scala [hunk 3]
```diff
@@ -36,3 +34,7 @@ trait TagClient[F[_]] {
   def remove(id: Int): F[Int]
 
 }
+
+object TagClient {
+  trait Handler[G[_]] extends TagClient[G]
```
Comment (diesalbla) [inline modules/examples/todolist/client/src/main/scala/clients/TagClient.scala]: Same as [here](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/examples/todolist/client/src/main/scala/clients/PingPongClient.scala [hunk 4]
```diff
@@ -17,9 +17,10 @@
...
 
-import freestyle.tagless.tagless
-
-@tagless(true)
 trait PingPongClient[F[_]] {
   def ping(): F[Unit]
 }
+
+object PingPongClient {
+  trait Handler[G[_]] extends PingPongClient[G]
```
Comment (diesalbla) [inline modules/examples/todolist/client/src/main/scala/clients/PingPongClient.scala]: Same as [here](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/examples/routeguide/client/src/main/scala/ClientProgram.scala [hunk 5]
```diff
@@ -23,14 +23,17 @@ import freestyle.tagless._
...
-@tagless(false)
 trait RouteGuideClient[F[_]] {
   def getFeature(lat: Int, lon: Int): F[Unit]
   def listFeatures(lowLat: Int, lowLon: Int, hiLat: Int, hiLon: Int): F[Unit]
   def recordRoute(features: List[Feature], numPoints: Int): F[Unit]
   def routeChat: F[Unit]
 }
 
+object RouteGuideClient {
+  trait Handler[G[_]] extends RouteGuideClient[G]
```
Comment (diesalbla) [inline modules/examples/routeguide/client/src/main/scala/ClientProgram.scala]: Same as [here](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/server/src/test/scala/protocol/Utils.scala [hunk 6]
```diff
@@ -165,6 +163,14 @@ object Utils extends CommonUtils {
       def bsws(eList: List[E]): F[E]
     }
 
+    object MyRPCClient {
+
+      trait Handler[G[_]] extends MyRPCClient[G]
```
Comment (diesalbla) [inline modules/server/src/test/scala/protocol/Utils.scala]: Same as [here](https://github.com/frees-io/freestyle-rpc/pull/296/files#r192694425).

File: modules/server/src/main/scala/GrpcServer.scala [hunk 7]
```diff
@@ -47,4 +47,36 @@ trait GrpcServer[F[_]] {
...
+    def awaitTerminationTimeout(timeout: Long, unit: TimeUnit): G[Boolean] =
+      fk(self.awaitTerminationTimeout(timeout, unit))
+
+    def awaitTermination(): G[Unit] = fk(self.awaitTermination)
+  }
+}
+
+object GrpcServer {
+
+  trait Handler[G[_]] extends GrpcServer[G]
```
[the remark this answers is not in the document]
Comment (pepegar) [reply modules/server/src/main/scala/GrpcServer.scala]: Cool, thanks for the tip! Changing it

**Pull request merged**